app/main.py

```diff
@@ -1,8 +1,39 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int | None) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    @classmethod
+    def create_or_get(cls, name: str, age: int = None) -> "Person":
+        if name in cls.people:
+            person_instance = cls.people[name]
+            person_instance.age = age
+        else:
+            person_instance = cls(name, age)
+        return person_instance
```

Comment:
Are you sure we need that check in this task? Try to rewrite without this function to simplify your code